### ISSUE TITLE
Compile error on self closing void elements

### ISSUE THREAD

**Issue by pozil** [salesforce/lwc#1483]
### Description
When compiling code that contains a self closing void element such as `<hr/>`, the compiler will throw an error.

#### Steps to Reproduce
Build a component with this template:
```html
<template>
    <hr/>
</template>
```

https://developer.salesforce.com/docs/component-library/tools/playground/rE1bVtKyj/1/edit

#### Expected Results
This compiles without error.

#### Actual Results
Compiler throws:
```
LWC1058: Invalid HTML syntax: non-void-html-element-start-tag-with-trailing-solidus. For more information, please visit https://html.spec.whatwg.org/multipage/parsing.html#parse-error-non-void-html-element-start-tag-with-trailing-solidus
```

Note that the spec linked in the error does mention that void elements can have a trailing solidus without any effect.

### Browsers Affected
n/a

### Version
Found this on in 1.1.0-alpha4 in a LWC OSS project.
Playground is affected as of 2019-08-30.

**Possible Solution**
Ignore void elements when applying compile time check as suggested in the HTML spec.

**Additional context/Screenshots**
This is particularly problematic because Prettier forces self closing tags on void elements and there's no way to disable that.
See: https://github.com/prettier/prettier/issues/5246

**Comment by muenzpraeger** [comment]
It's a Prettier error.

From the [spec](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hr).

`It must have start tag, but must not have an end tag.`

**Comment by pozil** [comment]
@muenzpraeger this is Mozilla documentation, not the spec.
This doc is wrong. The single source of truth is the [WHATWG spec](https://html.spec.whatwg.org/multipage/parsing.html#parse-error-non-void-html-element-start-tag-with-trailing-solidus) is what matters.

As linked in the compiler error:
> This error occurs if the parser encounters a start tag for an element that is not in the list of void elements or is not a part of foreign content (i.e., not an SVG or MathML element) that has a U+002F (/) code point right before the closing U+003E (>) code point. The parser behaves as if the U+002F (/) is not present.

and in the notes:

> It is also allowed for void elements, but doesn't have any effect in this case.

**Comment by muenzpraeger** [comment]
And for `hr` it says `No end tag`. ;-)

https://html.spec.whatwg.org/multipage/grouping-content.html#the-hr-element

**Comment by muenzpraeger** [comment]
I guess @diervo can judge best. He built the parser.

https://www.npmjs.com/package/parse5-with-errors

**Comment by pozil** [comment]
Adding [HTML element general spec](https://html.spec.whatwg.org/multipage/syntax.html#start-tags) for reference:

> 6. Then, if the element is one of the void elements, or if the element is a foreign element, then there may be a single U+002F SOLIDUS character (/). This character has no effect on void elements, but on foreign elements it marks the start tag as self-closing.

**Comment by pozil** [comment]
By the way, the `No end tag` info is  "A non-normative description" as per [Tag omission in text/html](https://html.spec.whatwg.org/multipage/dom.html#concept-element-tag-omission) link.

**Comment by caridy** [comment]
We rely on parse5 to parse the HTML files, that's why it is hard for us to fix this kind of issues.

**Comment by pozil** [comment]
Should anyone run into this issue where Prettier automatically adds a self-closing solidus`/`  character, the workaround is to manually exclude the problematic template file from Prettier processing.
To do that, add the path to the template file to a `.prettierignore` file at the root of the project.

**Comment by pozil** [comment]
Adding @pmdartus to the conversation since he 'owns' our private [parse5-with-errors](https://git.soma.salesforce.com/p-dartus/parse5-with-errors) fork of [parse5](https://github.com/inikulin/parse5).

@caridy I looked into parse5 and I believe (correct me if I'm wrong) that we introduced the error in parse5-with-errors. The current public version of parse5 is ok with self closing void elements.

We're missing a check for [void elements](https://html.spec.whatwg.org/multipage/syntax.html#void-elements) here:
https://git.soma.salesforce.com/p-dartus/parse5-with-errors/blob/master/lib/parser/index.js#L665

May I correct this with a PR?

As a separate note, @diervo the [parse5-with-errors NPM package public listing](https://www.npmjs.com/package/parse5-with-errors) is using a git soma URL as it's home page. We should probably fix that.

**Comment by caridy** [comment]
@pozil yes yes yes! go for it.

**Comment by pozil** [comment]
@pmdartus @caridy please review `p-dartus/parse5-with-errors/pull/2` (I can't assign reviewers).

**Comment by diervo** [comment]
We should certainly move that package from Pierre account xD

**Comment by caridy** [comment]
approved! if @apapko is good with it, we can move on!

**Comment by diervo** [comment]
@pozil yeah let's get @pmdartus to push a new version at lest roll.

**Comment by apapko** [comment]
approved! Thank you @pozil

**Comment by diervo** [comment]
Fix: https://github.com/salesforce/lwc/pull/1574

**Comment by apapko** [comment]
This has bee resolved with #1574 update of parse5-with-errors, which contains the self-closing tag [change](https://git.soma.salesforce.com/lwc/parse5-with-errors/pull/2).

**Comment by AllanOricil** [comment]
@diervo @apapko  I just got this error. What should I do to remove it?

![image](https://user-images.githubusercontent.com/55927613/133632958-b3a65a36-892a-474d-8f6e-093ffd2f0da7.png)
